client/state/sites/selectors.js

```diff
@@ -517,3 +518,95 @@ export function isCurrentSitePlan( state, siteId, planProductId ) {
 
 	return sitePlan.product_id === planProductId;
 }
+
+export function siteCanUpdateFiles( state, siteId ) {
+	if ( ! siteId ) {
+		return null;
+	}
+
+	if ( ! isJetpackSite( state, siteId ) ) {
+		return null;
+	}
+
+	if ( ! siteHasMinimumJetpackVersion( state, siteId ) ) {
+		return false;
+	}
+
+	const isMultiNetwork = getSiteOption( state, siteId, 'is_multi_network' );
+
+	if ( isMultiNetwork ) {
+		return false;
+	}
+
+	if ( ! isMainNetworkSite( state, siteId ) ) {
+		return false;
+	}
+
+	const fileModDisabled = getSiteOption( state, siteId, 'file_mod_disabled' );
+
+	if (
+		fileModDisabled &&
+		(
+			-1 < fileModDisabled.indexOf( 'disallow_file_mods' ) ||
+			-1 < fileModDisabled.indexOf( 'has_no_file_system_write_access' )
+		)
+	) {
+		return false;
+	}
+
+	return true;
+}
+
+export function isMainNetworkSite( state, siteId ) {
+	if ( ! siteId ) {
+		return null;
+	}
+
+	const site = getSite( state, siteId );
+	const isMultiNetwork = getSiteOption( state, siteId, 'is_multi_network' );
+
+	if ( isMultiNetwork ) {
+		return false;
+	}
+
+	if ( site.is_multisite === false ) {
+		return true;
+	}
+
+	if ( site.is_multisite ) {
+		const unmappedUrl = getSiteOption( state, siteId, 'unmapped_url' );
+		const mainNetworkSite = getSiteOption( state, siteId, 'main_network_site' );
+
+		if ( ! unmappedUrl || ! mainNetworkSite ) {
+			return false;
+		}
+
+		// Compare unmapped_url with the main_network_site to see if is the main network site.
+		return withoutHttp( unmappedUrl ) === withoutHttp( mainNetworkSite );
+	}
+
+	return false;
+}
+
+/**
```

Comment:
My initial reaction to this selector name was that it compares against a particular version, for which we already have a selector: `isJetpackMinimumVersion`. The names are very similar and I wonder if we could improve this to make it more clear that we're checking that the version is sufficient for Jetpack Manage support. Maybe `isSiteManageable` which returns `true` for WordPress.com sites and the result of the version check for Jetpack sites? In performing the version check, I'd also recommend making use of the existing `isJetpackMinimumVersion` selector.